sql-parser/src/main/antlr/SqlBase.g4

```diff
@@ -693,6 +695,10 @@ ESCAPE: 'ESCAPE';
 ASC: 'ASC';
 DESC: 'DESC';
 SUBSTRING: 'SUBSTRING';
+TRIM: 'TRIM';
```

Comment:
Can we mark these as non-reserved? Otherwise this is a breaking change.

Reply:
Dear me, I changed it in the docs and forgot to do so in the parser
good catch 👍

Reply:
@mfussenegger 
When I add the `TRIM` literal in the `nonReserved` List,  I get unknown function `trim('string')` when it is called with a single argument.

I noticed that other "special" functions are also not in the `nonReserved` List: `current_timestamp`, `case`, `extract`

When these are added, similar failures occur for these functions.

Reply:
If we can't find a way to make it work with it being `nonReserved` we have to add a note that `trim` is now a reserved keyword to the breaking changes.

Reply:
fixed with latest fixup